lib/Default/SmrPort.class.inc

```diff
@@ -813,11 +810,16 @@ class SmrPort {
 	public function getUpgrade() {
 		return $this->upgrade;
 	}
-	
+
+	public function getNumWeapons() {
+		return self::NUMBER_OF_PORT_WEAPONS;
+	}
+
 	public function &getWeapons() {
 		$weapons = array();
-		for($i=0;$i<self::NUMBER_OF_PORT_WEAPONS;++$i)
+		for ($i=0; $i<$this->getNumWeapons(); ++$i) {
```

Comment:
This function will be called once for every iteration of the loop, it's not a big deal because it's a cheap function but something to be aware in other situations/if the function becomes more expensive